host/sgx/switchless.c

```diff
@@ -82,52 +121,80 @@ static oe_result_t oe_stop_worker_threads(oe_switchless_call_manager_t* manager)
                 OE_RAISE(OE_THREAD_JOIN_ERROR);
     }
 
+    for (size_t i = 0; i < manager->num_enclave_workers; i++)
+    {
+        if (manager->enclave_worker_threads[i] != (oe_thread_t)NULL)
+            if (oe_thread_join(manager->enclave_worker_threads[i]))
+                OE_RAISE(OE_THREAD_JOIN_ERROR);
+    }
+
     result = OE_OK;
 done:
     return result;
 }
 
 oe_result_t oe_start_switchless_manager(
     oe_enclave_t* enclave,
-    size_t num_host_workers)
+    size_t num_host_workers,
+    size_t num_enclave_workers)
 {
     oe_result_t result = OE_UNEXPECTED;
     oe_result_t result_out = 0;
     oe_switchless_call_manager_t* manager = NULL;
-    oe_host_worker_context_t* contexts = NULL;
-    oe_thread_t* threads = NULL;
+    oe_host_worker_context_t* host_contexts = NULL;
+    oe_thread_t* host_threads = NULL;
+    oe_enclave_worker_context_t* enclave_contexts = NULL;
+    oe_thread_t* enclave_threads = NULL;
 
-    if (num_host_workers < 1 || enclave == NULL)
+    if (num_host_workers < 0 || num_enclave_workers < 0 || enclave == NULL)
```

Comment:
These are size_t and can never be < 0, correct?

Reply:
Good catch. I am surprised because I would expect the compiler to catch such checks.

We are compiling with `-Wextra` which detects and emits a `comparison of unsigned expression < 0 is always false` warning which we would have treated as error.

However, since December we are also compiling with `-Wno-type-limits` option which suppresses the above warning. I have created an issue #2766 to remove `-Wno-type-limits` option.

I will go ahead and remove these checks.